traffic_ops/traffic_ops_golang/wrappers.go

```diff
@@ -132,6 +122,93 @@ func wrapAccessLog(secret string, h http.Handler) http.HandlerFunc {
 	}
 }
 
+func wrapHeaders(h RegexHandlerFunc) RegexHandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request, p PathParams) {
+		w.Header().Set("Access-Control-Allow-Credentials", "true")
+		w.Header().Set("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept, Accept-Encoding, Set-Cookie, Cookie, Vary, Whole-Content-SHA512")
+		w.Header().Set("Access-Control-Allow-Methods", "POST,GET,OPTIONS,PUT,DELETE")
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("X-Server-Name", ServerName)
+		iw := &BodyInterceptor{w: w}
+		w = iw
+		h(w, r, p)
+		sha := sha512.Sum512(iw.body)
+		w.Header().Set("Whole-Content-SHA512", base64.StdEncoding.EncodeToString(sha[:]))
+
+		//TODO: - drichardson - hook in
+		//gzipHandler(w, r, iw.body)
+		//fmt.Printf("outbound headers ---> %v\n", w.Header())
+
+	}
+}
+
+// gzipHandler takes a function which cannot error and returns only bytes, and wraps it as a http.HandlerFunc. The errContext is logged if the write fails, and should be enough information to trace the problem (function name, endpoint, request parameters, etc).
+//TODO: drichardson - refactor these to a generic area
+func gzipHandler(w http.ResponseWriter, r *http.Request, bytes []byte) {
+
+	bytes, err := gzipIfAccepts(r, w, bytes)
+	if err != nil {
+		log.Errorf("gzipping request '%v': %v\n", r.URL.EscapedPath(), err)
+		code := http.StatusInternalServerError
+		w.WriteHeader(code)
+		if _, err := w.Write([]byte(http.StatusText(code))); err != nil {
+			log.Warnf("received error writing data request %v: %v\n", r.URL.EscapedPath(), err)
+		}
+		return
+	}
+
+	w.Write(bytes)
+
+	log.Write(w, bytes, r.URL.EscapedPath())
```

Comment:
The `log.Write` writes, and logs if an error occurs. So with the previous `w.Write`, this is writing twice.

I know, it needs a better name x_x